### PR TITLE
✨ Allowing mutability for the PreKubeadmCommands, PostKubeadmCommands and Files

### LINKED ISSUE #3014: Possibility to modify preKubeadmCommands during a KCP update
**User Story**

As a developer I would like to have a possibility to modify preKubeadmCommands during KCP update for maintaining compatibility with new images.

**Detailed Description**

Situation can occur if we need to do something different due to new versions of the kubernetes binaries (kubeadm, kubelet, kubelet) that are baked into a new image.
E.g change a load balancer configuration, call another sysctl to start a new service or something else that was moved or renamed.

/kind feature

### PR DESCRIPTION
**Pull request by furkatgofurov7**
What this PR does / why we need it:
Allows to modify preKubeadmCommands, postKubeadmCommands and files in KCP

Which issue(s) this PR fixes:
Fixes #3014

### CODE REVIEW

**Comment by vincepri** [comment]
@furkatgofurov7 Need to sign the CLA

**Comment by detiber** [comment]
lgtm pending CLA validation and tests

Might be good to get some feedback from @rudoi and @sethp-nr, since I believe they were also interested in additional mutability of KCP fields.

**Comment by furkatgofurov7** [comment]
"I signed it"

**Comment by furkatgofurov7** [comment]
@vincepri @detiber signed the CLA.

File: controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook.go [hunk 1]
```diff
@@ -74,6 +74,9 @@ const (
 	spec                 = "spec"
 	kubeadmConfigSpec    = "kubeadmConfigSpec"
 	clusterConfiguration = "clusterConfiguration"
+	preKubeadmCommands   = "preKubeadmCommands"
+	postKubeadmCommands  = "postKubeadmCommands"
+	files                = "files"
```
Comment (fabriziopandini) [inline controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook.go:79]: The above 3 field names are used only in one place, so I don't think it is required a constant (and for consistency, there are no constant for many other fields with the exception of top level entries)
So I IMO this should be removed

Comment (furkatgofurov7) [reply controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook.go:79]: @fabriziopandini thanks for review, and to add, these 3 field names are in the same level as `clusterConfiguration`, which has a constant as well as for consistency.

Comment (fabriziopandini) [reply controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_webhook.go:79]: Ok, this makes sense in its way

**Pull request merged**